python/mpt/src/mpt/ethereum_tries.py

```diff
@@ -71,3 +108,181 @@ def from_data(data: Dict[str, Any]):
+            )
+            current_path = current_path + node.key_segment
+
+            # subnode is a hash, so we need to resolve it
+            if len(node.subnode) == 32:
+                next_node = self.nodes.get(node.subnode)
+                if next_node is None:
+                    # If the subnode is not found, we assume this path
+                    # is not needed for block execution
+                    return
+                self.traverse_trie(next_node, current_path, process_leaf, **kwargs)
+                return
+            # if subnode is less than 32 bytes, it's an embedded (RLP-encoded) node
+            if len(node.subnode) < 32:
+                next_node = decode_node(node.subnode)
+                self.traverse_trie(next_node, current_path, process_leaf, **kwargs)
+                return
+
+            if len(node.subnode) > 32:
+                raise ValueError(f"Invalid subnode length: {len(node.subnode)}")
+
+        if isinstance(node, LeafNode):
+            logger.debug(
+                f"Traversing leaf node with current path 0x{nibble_list_to_bytes(current_path + node.rest_of_key).hex()}"
+            )
+            full_path = nibble_list_to_bytes(current_path + node.rest_of_key)
+            return process_leaf(node, full_path, **kwargs)
+
+        return
+
+    def process_account_leaf(
+        self,
+        node: LeafNode,
+        full_path: Bytes,
+        state: State,
+    ) -> None:
+        logger.debug(f"Processing account leaf node with path 0x{full_path.hex()}")
+        address = self.address_preimages.get(full_path)
+        if address is None:
+            logger.debug(
+                f"Address not found in address preimages: {full_path}, skipping"
+            )
+            return
+
+        account_node = AccountNode.from_rlp(node.value)
+        account_code = self.get_code(account_node.code_hash, address)
+        account = account_node.to_account(account_code)
+
+        logger.debug(
+            f"Setting account 0x{address.hex()} with nonce {account.nonce}, balance {account.balance}, code hash 0x{keccak256(account.code).hex()}"
+        )
+        set_account(state, address, account)
+
+        if account_node.storage_root == EMPTY_TRIE_HASH:
+            logger.debug(f"Storage root is empty for account {address.hex()}, skipping")
+            return
+
+        # We need to resolve the storage root of the account
+        storage_root_node = self.nodes.get(account_node.storage_root)
+        if storage_root_node is None:
+            logger.debug(
+                f"Storage root node not found for account 0x{address.hex()}, skipping"
+            )
+            return
+        logger.debug(
+            f"Storage root node found for 0x{address.hex()}, opening storage trie"
+        )
+        self.resolve_storage(storage_root_node, b"", state, address)
+
+    def process_storage_leaf(
```

Comment:
```suggestion
    def set_storage_from_leaf(
```

nit: function name should indicate mutation of state